client/remoterepo.c

```diff
@@ -306,7 +807,7 @@ TDNFDownloadFile(
                 fclose(fp);
                 fp = NULL;
             }
-            dwError = TDNFParseAndGetURLFromMetalink(pTdnf, pszRepo, pszFile);
+            dwError = TDNFParseAndGetURLFromMetalink(pTdnf, pszRepo, pszFile, &ml_file);
```

Comment:
Why do we need to keep passing address of ml_file and use double and triple pointers? Why can we use single pointer throughout

Reply:
I thought this is better to track, let me know your thoughts?

Reply:
I say, it makes life difficult. We should avoid using double and triple pointers unless it's absolutely necessary.